### PR TITLE
(feat) Centralize ErrorState component

### PR DESCRIPTION
**Pull request by ZacButko**
## Requirements
- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.

#### For changes to apps
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).

#### If applicable
- [x] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
This takes the ErrorState component, which is shown when data loading fails, out of esm-patient-common-lib and makes it more broadly accessible in esm-framework by way of esm-styleguide. The component itself is copy pasted. Everything else is support to get the component to render and test in the core package

## Screenshots
Same UI as was in patient chart
Screen Shot 2022-07-08 at 8 00 04 AM

## Other
This took quite a bit more work than expected. Had to install Jest and testing library for jest, this version bump caused some changes in default values, notably for testEnvironment from jsdom to node, hence the addition of jest.config.json and explicitly stating jsdom. Next test environment needed to be added as dev dependency, but this is broken with v13 for React < 18, so had to pin this to ^12.x. Next translations weren't implemented yet, so this was solved by adding __mocks__/react-i18next.js with standard mock info here.

### CODE REVIEW

**Review by denniskigen — APPROVED** [review]
Thanks, @ZacButko. Sorry for having to go through all the pain.

File: packages/framework/esm-styleguide/src/error-state/error-state.component.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === "tablet";
+
+  return (
+    <Tile light className={styles.tile}>
+      <div className={isTablet ? styles.tabletHeading : styles.desktopHeading}>
+        <h4>{headerTitle}</h4>
+      </div>
+      <p className={styles.errorMessage}>
+        {t("error", "Error")} {`${error?.response?.status}: `}
```
Comment (ibacher) [inline packages/framework/esm-styleguide/src/error-state/error-state.component.tsx:25]: Not a problem introduced with this PR, but I'd be curious to know what the strategy is for translating strings like this... AFAICT, there are no real translations for this anywhere...

**Review by ibacher — APPROVED** [review]

File: packages/framework/esm-styleguide/src/error-state/error-state.component.tsx (hunk 1 above)
Comment (ZacButko) [reply packages/framework/esm-styleguide/src/error-state/error-state.component.tsx:25]: The translation function only translates the word "Error". The response status I'm assuming is the status code which should be a number, unless we want to translate way from Arabic numerals? As for the `response.statusText` is it reasonable to say that this is the backend's responsibility to translate?

Comment (ibacher) [reply packages/framework/esm-styleguide/src/error-state/error-state.component.tsx:25]: I think you missed my point: Say I want to say that "Error" in French should be "Erreur". Where do I put that string?

Comment (ZacButko) [reply packages/framework/esm-styleguide/src/error-state/error-state.component.tsx:25]: Do you just mean that the translation file is missing?

**Pull request merged**